src/catalogue.js

```diff
@@ -92,17 +77,21 @@ export function loadItems(projectId, options = {}) {
         ...REQUEST_DEFAULTS,
         ...options,
         ...getRequiredDataSets(options)
-    }, ['dataSetIdentifier', 'returnAllDateDataSets']);
-
-    let bucketItems = get(cloneDeep(options), 'bucketItems.buckets');
-    if (bucketItems) {
-        bucketItems = bucketItemsToExecConfig(bucketItems);
-        return loadCatalog(
-            projectId,
-            {
+    }, [
+        'dataSetIdentifier',
+        'returnAllDateDataSets',
+        'attributesMap'
+    ]);
+
+    const mdObj = get(cloneDeep(options), 'bucketItems');
+    const { attributesMap } = options;
+    const hasBuckets = get(mdObj, 'buckets') !== undefined;
```

Comment:
Lodash has function `has` that does exactly this

Reply:
Careful - `has` do return true when `mdObj = { buckets: undefined }`

Reply:
rewritten to cover all bad cases